File: Java/49.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> gans = new HashMap();
        
        for(String s: strs){
            char[] alpha = new char[26];
            for(char c: s.toCharArray()){
                alpha[c-'a']++;
            }
            String str=new String(alpha);
            
            if(gans.get(str)==null){
                gans.put(str, new ArrayList<>());
            }
            gans.get(str).add(s);
        }
        List<List<String>> res = new ArrayList<>();
        res.addAll(gans.values());
        return res;
    }
}

File: C++/49.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<vector<int>, vector<string>> gans;
        
        for (string str : strs)
        {
            vector<int> alpha(26);
            for(char ch : str)
            {
                alpha[ch - 'a'] += 1;
            }
            gans[alpha].push_back(str);
        }
        
        vector<vector<string>> result;
        
        for(map<vector<int>, vector<string>>::iterator it = gans.begin(); it != gans.end(); ++it)
        {
            result.push_back(it->second);
        }
        return result;
    }
};